addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -3678,6 +3688,154 @@ options.registry.WebsiteAnimate = options.Class.extend({
     },
 });
 
+/**
+ * Allows edition of text "Highlight Effects" following this generic structure:
+ * `<span class="o_text_highlight">
+ *      <span class="o_text_highlight_item">
+ *          line1-textNode1 [line1-textNode2,...]
+ *          <svg.../>
+ *      </span>
+ *      [<br/>]
+ *      <span class="o_text_highlight_item">
+ *          line2-textNode1 [line2-textNode2,...]
+ *          <svg.../>
+ *      </span>
+ *      ...
+ * </span>`
+ * To correctly adapt each highlight unit when the text content is changed.
+ */
+options.registry.TextHighlight = options.Class.extend({
+    custom_events: Object.assign({}, options.Class.prototype.custom_events, {
+        "user_value_widget_opening": "_onWidgetOpening",
+    }),
+    /**
+     * @override
+     */
+    async start() {
+        await this._super(...arguments);
+        this.leftPanelEl = this.$overlay.data("$optionsSection")[0];
+        // Reduce overlay opacity for more highlight visibility on small text.
+        this.$overlay[0].style.opacity = "0.25";
+        this.$overlay[0].querySelectorAll(".o_handle")
+            .forEach(handle => handle.classList.add("pe-none"));
+    },
+    /**
+     * Move "Text Effect" options to the editor's toolbar.
+     *
+     * @override
+     */
+    onFocus() {
+        this.options.wysiwyg.toolbarEl.append(this.$el[0]);
+    },
+    /**
+     * @override
+     */
+    onBlur() {
+        this.leftPanelEl.appendChild(this.el);
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Activates & deactivates the text highlight effect.
+     *
+     * @see this.selectClass for parameters
+     */
+    async setTextHighlight(previewMode, widgetValue, params) {
```

Comment:
It shouldn't have the same name as the other `setTextHighlitght` function (in text_processing.js).. it's confusing ;)